end_to_end_test/lib/values.g.dart

```diff
@@ -1163,6 +1222,118 @@ class CompoundValueNoNestingBuilder
+      : super._() {
+    if (simpleValue == null) {
+      throw new BuiltValueNullFieldError(
+          'CompoundValueComparableBuilders', 'simpleValue');
+    }
+  }
+
+  @override
+  CompoundValueComparableBuilders rebuild(
+          void updates(CompoundValueComparableBuildersBuilder b)) =>
+      (toBuilder()..update(updates)).build();
+
+  @override
+  CompoundValueComparableBuildersBuilder toBuilder() =>
+      new CompoundValueComparableBuildersBuilder()..replace(this);
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(other, this)) return true;
+    return other is CompoundValueComparableBuilders &&
+        simpleValue == other.simpleValue &&
+        validatedValue == other.validatedValue;
+  }
+
+  @override
+  int get hashCode {
+    return $jf($jc($jc(0, simpleValue.hashCode), validatedValue.hashCode));
+  }
+
+  @override
+  String toString() {
+    return (newBuiltValueToStringHelper('CompoundValueComparableBuilders')
+          ..add('simpleValue', simpleValue)
+          ..add('validatedValue', validatedValue))
+        .toString();
+  }
+}
+
+class CompoundValueComparableBuildersBuilder
+    implements
+        Builder<CompoundValueComparableBuilders,
+            CompoundValueComparableBuildersBuilder> {
+  _$CompoundValueComparableBuilders _$v;
+
+  SimpleValue _simpleValue;
+  SimpleValue get simpleValue => _$this._simpleValue;
+  set simpleValue(SimpleValue simpleValue) => _$this._simpleValue = simpleValue;
+
+  ValidatedValue _validatedValue;
+  ValidatedValue get validatedValue => _$this._validatedValue;
+  set validatedValue(ValidatedValue validatedValue) =>
+      _$this._validatedValue = validatedValue;
+
+  CompoundValueComparableBuildersBuilder();
+
+  CompoundValueComparableBuildersBuilder get _$this {
+    if (_$v != null) {
+      _simpleValue = _$v.simpleValue;
+      _validatedValue = _$v.validatedValue;
+      _$v = null;
+    }
+    return this;
+  }
+
+  @override
+  void replace(CompoundValueComparableBuilders other) {
+    if (other == null) {
+      throw new ArgumentError.notNull('other');
+    }
+    _$v = other as _$CompoundValueComparableBuilders;
+  }
+
+  @override
+  void update(void updates(CompoundValueComparableBuildersBuilder b)) {
+    if (updates != null) updates(this);
+  }
+
+  @override
+  _$CompoundValueComparableBuilders build() {
+    final _$result = _$v ??
+        new _$CompoundValueComparableBuilders._(
+            simpleValue: simpleValue, validatedValue: validatedValue);
+    replace(_$result);
+    return _$result;
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(other, this)) return true;
+    return other is CompoundValueComparableBuildersBuilder &&
+        simpleValue == other.simpleValue &&
+        validatedValue == other.validatedValue;
+  }
+
+  @override
+  int get hashCode {
+    return $jf($jc($jc(0, simpleValue.hashCode), validatedValue.hashCode));
```

Comment:
Same hashcode as the Built, but not equal.
Should it have a different seed?

Reply:
Done, and added unit test.